// @webpack-cli/init/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const generators_1 = require("@webpack-cli/generators");
const utils_1 = require("@webpack-cli/utils");
class InitCommand {
    async apply(cli) {
        await cli.makeCommand({
            name: 'init [scaffold...]',
            alias: 'c',
            description: 'Initialize a new webpack configuration.',
            usage: '[scaffold...] [options]',
            pkg: '@webpack-cli/init',
        }, [
            {
                name: 'auto',
                type: Boolean,
                description: 'To generate default config',
            },
            {
                name: 'force',
                type: Boolean,
                description: 'To force config generation',
            },
            {
                name: 'generation-path',
                type: String,
                description: 'To scaffold in a specified path',
            },
        ], async (scaffold, program) => {
            const options = program.opts();
            if (scaffold && scaffold.length > 0) {
                await utils_1.npmPackagesExists(scaffold);
                return;
            }
            utils_1.modifyHelperUtil('init', generators_1.initGenerator, null, null, options.auto, options.force, options.generationPath);
        });
    }
}
exports.default = InitCommand;
